frontend/components/redeem/RedeemFormButtons.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  progressButtonContainerSx,
+  progressButtonFirstStepContainerSx,
+} from '../../styles/components/redeem/RedeemStyles';
+import Button from '../generic/Button';
+import { Stack } from '@mui/system';
+
+interface Props {
+  activeStep: number;
+  setActiveStep: (value: ((prevState: number) => number) | number) => void;
+  minStep: number;
+  maxStep: number;
+  handleClickBack?: () => void;
+  handleClickNext?: () => void;
+  shouldDisablePrimaryButton?: boolean;
+}
+
+const RedeemFormButtons = ({
```

Comment:
So why do you feel that way? If you can explain it or have a concrete suggestion then maybe I'll change it in the next pr. I don't like how I'm currently passing the state down more than 1 layers also, but I think most of the responsibility of changing steps should be in the form navigation buttons.

Otherwise, I can put all the functions in the index.tsx page and put a lot of if-elses there of what to do for clicking back or next. But is that any better, we're making index.tsx bloated with more responsibilities? Then what's the point of creating a form navigation component to handle that responsibility? Also, right now each step has slightly different things to do on each navigation and only they know what should happen, so that's why I let them handle the state.

Anyway I'll think about it